### ISSUE TITLE
[3.10 to 4] Pre-Update Check shows "Update Information Unavailable" for "Installer - Install from Web" plugin

### ISSUE THREAD

**Issue by ReLater** [joomla/joomla-cms#35103]
### Steps to reproduce the issue
- Tested an Update from 3.10.0-rc3-dev current nightly to 4.0.0-rc7-dev current nighly.
- Get a warning in Pre-Update Checker that "Installer - Install from Web" plugin has no update informations.

![12-08-_2021_17-52-44](https://user-images.githubusercontent.com/20780646/129229613-1ddff050-6ffb-419b-b20f-8d7f9da84634.jpg)

- As far as I know there are no issues when migrating with this message.
- Maybe the plugin should just provide an OK for Joomla 4 in https://appscdn.joomla.org/webapps/jedapps/webinstaller.xml ????????????????????????????

**Comment by zero-24** [comment]
IIRC this has been converted into an core extension with Joomla 4 so there is no dedicated 4.x version. I have not checked yet whether the upgrade just replaces the old version so it could be filtered out here or is defined to be compatible with 4.0 in the update server.

**Comment by ReLater** [comment]
I'm checking at the moment what happens when I ignore the message.

**Comment by ReLater** [comment]
- Before migration: version 2.1.2 with extension_id 10003.
- After: version 1.1.1 (which is a bit weird but I don't care) with extension_id 10039.
- - Old plugin 10003 is uninstalled from db
- - old update server is removed from db
- - plugin works

**Comment by zero-24** [comment]
Ok lets go and update the update server: https://appscdn.joomla.org/webapps/jedapps/webinstaller.xml to "claim" that 2.1.2 is supported in 4.0 too. It will be replaced by the update anyway and filtering it out would just add confusion i guess, now we have to find out where the file can be found and updated. Maybe @rdeutz does know which server this file comes from? Its not part of the repo: https://github.com/joomla-extensions/install-from-web-client

> After: version 1.1.1 (which is a bit weird but I don't care) with extension_id 10039.

I guess the Core version should just be 4.0.0 right @wilsonge ?

**Comment by zero-24** [comment]
Found PR done here: https://github.com/joomla/install-from-web-server/pull/59

**Comment by ReLater** [comment]
> I guess the Core version should just be 4.0.0 right

<strike>I think it must be higher than 2.1.2 at least. Otherwise you'll get an update message in Joomla 4 with changes in https://github.com/joomla/install-from-web-server/pull/59</strike>

**Comment by zero-24** [comment]
Didnt you said the update server is gone on Upgrade to 4.0.0

**Comment by ReLater** [comment]
You're right

**Comment by zero-24** [comment]
The update server has been updated so i would like to close here ok @ReLater ?

**Comment by ReLater** [comment]
With tears in my eyes I see an update message in Joomla 4 for version 1.1.1 to 2.1.2.
With an update server that can't be found in "Extensions: Update Sites".
When I click "Rebuild" there the update message is gone.
I'm really confused and sorry that I gave wrong informations above(???).

So I repeat your question to @wilsonge : "I guess the Core version should be 4.0.0 right?".

**Comment by ReLater** [comment]
@zero-24 @beat 

In https://appscdn.joomla.org/webapps/jedapps/webinstaller.xml:

Can we change the block with `<targetplatform name="joomla" version="4.*"/>` to `<version>1.1.1</version>` or lower (some dummy nonsense) instead now `<version>2.1.2</version>`? 

i think that is "easier" (2 days before J4 stable) than changing the webinstaller version in J!4 core.

**Comment by ReLater** [comment]
But ???????????????????????????????????????????????? By now I am too confused and tired to understand...

**Comment by zero-24** [comment]
> i think that is "easier" (2 days before J4 stable) than changing the webinstaller version in J!4 core.

the cleanest way would be to update the version shipped with the CMS to 4.0.0 changing the version on the update server XML can be done too but changing the version in the CMS should and can be done for 4.0.0 anyway. I can prepare the PR later.

**Comment by zero-24** [comment]
PR: https://github.com/joomla/joomla-cms/pull/35127

**Comment by zero-24** [comment]
> Can we change the block with <targetplatform name="joomla" version="4.*"/> to <version>1.1.1</version> or lower (some dummy nonsense) instead now <version>2.1.2</version>?

Well that would mean we claim that only a lower version than the installed right now is compatible with 4.0 and that will only confuse people cause that would show up as "updates required" thats why i used the last stable to claim 4.0 compatibility.

**Comment by zero-24** [comment]
Closing as there is a PR #35127
